### ISSUE TITLE
Open FtpWebRequest for customization

### ISSUE THREAD

**Issue by magol** [dotnet/corefx#17030]
`FtpWebRequest` works great when communicating with "normal" FTP servers. But if I try to communicate with more odd FTP servers, it becomes a problem when there is not much opportunity to adjust `FtpWebRequest`.

As an example, we have at work an old FTP server with the following characteristics:
* There is no proper directory structure as it works on Unix/Dos (and therefore no root). We specifying the path with parameters like this:  `CWD ~TNA=AMP,VNA=VOL03,FNA=U-ED-B2-USL`

* If you make a listing of files without specifying `FNA`, we get a tree structure with everything on the entire partition.

I do not expect that this odd behavior to be implemented in `FtpWebRequest`, but I wish that there is any why to customize `FtpWebRequest `to handle these and other special cases.

One way is to have a callback just before commands are sent to the FTP server to provide the ability to customize what is sent and then another callback when the response arrives. Another way might be to make it possible to inherit from `FtpWebRequest `to make your own customized version.

**Comment by davidsh** [comment]
We consider FtpWebRequest one of the more legacy APIs originally from the .NET Framework.  We ported it to .NET Core to assist devs when porting their existing code to .NET Core.  

As you have observed, there are limitations to the current design of FtpWebRequest.  The design pattern it uses is inherited from the WebRequest class.  If we were re-doing this feature, we would create a different type of object model and have a class called 'FtpClient'.  Then there would be methods/properties in a pattern similar to TcpClient which would allow more customization of verbs and other FTP-specific commands.

So, bottom line, we think it is more useful for a new "FTP" class to be created with a fresh object model than to try to re-define the existing FtpWebRequest class.

cc: @karelz @stephentoub

**Comment by magol** [comment]
I understand that you not want to make a major change of `FtpWebequest`. The question is if it is possible to make a smaller fix that make this possible.

[Matt Mitchell](http://mattmitchell.com.au/how-to-change-the-commands-ftpwebrequest-sends/) suggests to use reflection to do this. Maybe we can make this possible without reflections?

**Comment by karelz** [comment]
I would be fine with minor fix contributions to legacy APIs. Of course, it would require clear non-breaking behavior, test coverage, etc.
Your comment above hints this would require a new API. That however, does not fit "minor fix" category.

**Comment by karelz** [comment]
Closing as we do not plan to add the functionality. If there is more interest in future, we can revisit.
